### PR TITLE
Add Gene-MTEB tasks

### PR DESCRIPTION
**Pull request by shangshang-wang**
### Code Quality

- [x] **Code Formatted**: Format the code using `make lint` to maintain consistent style.

### Documentation

- [x] **Updated Documentation**: Add or update documentation to reflect the changes introduced in this PR.

### Testing

- [x] **Tests Passed**: Run tests locally using `make test` or `make test-with-coverage` to ensure no existing functionality is broken.

### Adding datasets checklist

**Reason for dataset addition**: To add specific benchmarks tailored to metagenomic analysis, where tasks involve classification, multi-label classification and clustering based on public available metagenomic data samples. 

- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`.

### CODE REVIEW

File: mteb/tasks/Classification/metagene/HumanMicrobiomeProjectDemonstrationClassification.py [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+class HumanMicrobiomeProjectDemonstrationClassificationDisease(AbsTaskClassification):
+    metadata = TaskMetadata(
+        name="HumanMicrobiomeProjectDemonstrationClassificationDisease",
+        description="A Classification task for predicting the disease from the Human Microbiome Project Demonstration dataset.",
+        reference="https://huggingface.co/datasets/metagene-ai/HumanMicrobiomeProjectDemonstration/tree/main/hmpd/disease",
+        dataset={
+            "path": "metagene-ai/HumanMicrobiomeProjectDemonstration",
+            "name": "disease",
+            "revision": "main",
```
Comment (Samoed) [inline mteb/tasks/Classification/metagene/HumanMicrobiomeProjectDemonstrationClassification.py:15]: You need to specify exact revision of your dataset

File: README.md
Comment (Samoed) [inline README.md:1]: I think this is from your fork. Please update the README.

File: mteb/tasks/Classification/metagene/HumanMicrobiomeProjectDemonstrationClassification.py [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+        split_datasets = full_train_dataset.train_test_split(
+            test_size=test_size,
+            shuffle=True,
+            seed=42)
+        new_train_dataset = split_datasets['train']
+        new_test_dataset = split_datasets['test']
+        self.dataset = datasets.DatasetDict({
+            'train': new_train_dataset,
+            'test': new_test_dataset
+        })
```
Comment (Samoed) [inline mteb/tasks/Classification/metagene/HumanMicrobiomeProjectDemonstrationClassification.py:80]: Can you upload dataset with these splits created directly?

Comment (Samoed) [reply mteb/tasks/Classification/metagene/HumanMicrobiomeProjectDemonstrationClassification.py:80]: Or can you create function in this file that would be used in all tasks

File: mteb/tasks/Classification/metagene/HumanMicrobiomeProjectDemonstrationClassification.py [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+            author={Liu, Ollie and Jaghouar, Sami and Hagemann, Johannes and Wang, Shangshang and Wiemels, Jason and Kaufman, Jeff and Neiswanger, Willie},
+            journal={arXiv preprint arXiv:2501.02045},
+            year={2025}
+        }
+        """
+    )
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.method = "logReg"
```
Comment (Samoed) [inline mteb/tasks/Classification/metagene/HumanMicrobiomeProjectDemonstrationClassification.py:42]: Your tasks can be run with only a logreg?

File: mteb/tasks/Classification/metagene/HumanMicrobiomeProjectDemonstrationClassification.py [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+
+    def load_data(self, **kwargs):
+        if self.data_loaded:
+            return
+
+        from transformers.trainer_utils import set_seed
```
Comment (Samoed) [inline mteb/tasks/Classification/metagene/HumanMicrobiomeProjectDemonstrationClassification.py:48]: Can you move imports to start of file?

File: mteb/tasks/Clustering/metagene/HumanMicrobiomeProjectReferenceClustering.py [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+            test_size=test_size,
+            shuffle=True,
+            seed=42)
+        new_train_dataset = split_datasets['train']
+        new_test_dataset = split_datasets['test']
+        self.dataset = datasets.DatasetDict({
+            'train': new_train_dataset,
+            'test': new_test_dataset
+        })
+        print(f"\nSplitting the data with test_size={test_size}")
```
Comment (Samoed) [inline mteb/tasks/Clustering/metagene/HumanMicrobiomeProjectReferenceClustering.py:71]: Please, remove prints or change to logging

**Review by Samoed — COMMENTED** [review]
Great additions!

File: mteb/tasks/Classification/metagene/HumanMicrobiomeProjectDemonstrationClassification.py [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+        description="A Classification task for predicting the disease from the Human Microbiome Project Demonstration dataset.",
+        reference="https://huggingface.co/datasets/metagene-ai/HumanMicrobiomeProjectDemonstration/tree/main/hmpd/disease",
+        dataset={
+            "path": "metagene-ai/HumanMicrobiomeProjectDemonstration",
+            "name": "disease",
+            "revision": "main",
+        },
+        type="Classification",
+        category="s2s",
+        modalities=["text"],
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/metagene/HumanMicrobiomeProjectDemonstrationClassification.py:19]: A text modality seems someone misleadering. It input is a string of base pairs right?

Comment (KennethEnevoldsen) [reply mteb/tasks/Classification/metagene/HumanMicrobiomeProjectDemonstrationClassification.py:19]: keeping "text" here might be fine but it should be clear for the description. I would also make it clear in the task subtypes.

**Comment by KennethEnevoldsen** [comment]
@shangshang-wang seems like this issue has gotten stale - we would be more than happy to get this merged in if you have the time to adress the comments

**Comment by KennethEnevoldsen** [comment]
Will close this as it is stale
